engine/src/main/java/org/terasology/rendering/dag/RenderGraph.java

```diff
@@ -28,11 +30,22 @@ public RenderGraph() {
         nodes = Lists.newArrayList();
     }
 
-    public String addNode(Node node, String suggestedId) {
+    public String addNode(Node node, String suggestedId) { // TODO: Change suggestedId to SimpleUri
```

Comment:
Ah, I left this as a string because there was a lot of grunt work, and I wanted to be sure I was going down the right path before I commited that much work on this.
( Weirdly enough, I'm fine with redoing entire systems, but not the grunt work :P )